pkg/ccl/streamingccl/streamingest/stream_ingestion_planning.go

```diff
@@ -174,6 +174,14 @@ func ingestionPlanHook(
 				oldTenantID.ToUint64(), newTenantID.ToUint64())
 		}
 
+		// Create a new tenant for the replication stream
+		if _, err := sql.GetTenantRecord(ctx, p.ExecCfg(), nil, newTenantID.ToUint64()); err == nil {
+			return errors.Newf("tenant with id %s already exists", newTenantID)
+		}
+		if err := p.ExtendedEvalContext().Tenant.CreateInactiveTenant(ctx, newTenantID.ToUint64()); err != nil {
```

Comment:
Coming back to this. I think in the long run we want to do something a little different.

The problem with initialising the tenant is that it means that the keyspace we are writing into isn't completely empty.  Namely it will have entries schema entries for the default database, the cluster version setting, the id sequence populated.  If this destination tenant is created after the source tenant, then the timestamp on those KVs will likely be above whatever gets replicated to us.  

> because it seemed like just creating a record did eventually initialize the rest of the tenant such that I run queries against it

Did you see this outside of the streaming replication tests? If you only saw it in the streaming replication tests, my guess for what you were seeing is that eventually enough of the _source_ cluster's tenant data is replicated to it for it to be viable.

Reply:
I think your theory on "the tenant works when enough has been streamed in" is true, since attempting to connect right after stream creation doesn't work, connecting after waiting 2 seconds but without cutover still works, and connecting after cutover always works.